### DISCUSSION TITLE
Does this work with Message in iCloud

### DISCUSSION THREAD

**Discussion by the-guitar-man** [ReagentX/imessage-exporter#290] [Q&A] [answered]
Sorry for such a basic question, but I am in bit of conundrum. I have been using Messages and iMessage for years, with it set to never delete messages, and I do NOT have Messages in iCloud enabled. So I believe that each of my devices stores its own copy of every message, and each device has 30GB of message content.

My question is, if I enable Message in iCloud on my Mac, will that make it harder to do a complete backup of my messages? Right now, my device has a local copy of every message, so I worry that if I enable this, I will have to scroll to the beginning of every conversation or something to get local copies to export. I was hoping to do a backup before enabling this, but ironically, I do not have the hard drive space to install xcode command line tools (10GB) to get imessage_explorer installed and running. 

Thanks for any help.

**Comment by ReagentX** [comment, marked as answer]
> I enable Message in iCloud on my Mac, will that make it harder to do a complete backup of my messages?

It is hard to say anything for certain about iCloud. I would export both your macOS and iOS messages using `imessage-exporter` before you enable iCloud just to have a copy. Right now you know you have a complete set of data and I wouldn't change anything until you have backed it up somehow. That said, I use messages in iCloud and all of my messages still get synced to both my phone and my laptop; I have it set to store messages indefinitely. 

If you don't have 10GB of drive space available it's also unlikely you would be able run `imessage-exporter` with any of the [`--copy-method`](https://github.com/ReagentX/imessage-exporter/blob/develop/imessage-exporter/README.md#how-to-use)s enabled in the first place, because attachments can take up a lot of space (and your post suggests approximately 30GB).

You shouldn't need to install the dev tools to run `imessage-exporter`, you can download a [prebuilt release](https://github.com/ReagentX/imessage-exporter/blob/develop/imessage-exporter/README.md#prebuilt-binaries).

**Reply by the-guitar-man** [reply]
Thanks for the info. I took your advice and did backups before doing anything. 

In regards to storage space, you are correct, with almost no space left, things get tricky. For anyone who is in a similar situation, I ended up buying an external SSD and creating a Symlink to a backup folder on the external, and then moving that to the normal backup locaiton, allowing me to backup to my external disk. A quick google search can guide anyone through the process, so I will not detail it here.

Will follow up once I have backed everything up and moved to iCloud, will see if I can still back everything up after the switch.

**Reply by ReagentX** [reply]
You can specify the output folder to be on an external disk:

- [Instructions](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#how-to-use)
- [Examples](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#examples)

**Reply by the-guitar-man** [reply]
Thanks! That works for the imessage exporter output, but I was referring to the device backups that may be required before using imessage exporter.
